File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> vis(nums.begin(), nums.end());
        int maxlen = 0;
        for(int& num:nums){
            int count = 0;
            if(!vis.count(num - 1)){
                while(vis.count(num)){
                    count++;
                    num++;
                }
            }
            
            maxlen = max(maxlen, count);
        }
        
        return maxlen;
    }
};

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        ans = 0
        n_set = set(nums)
        
        for i in nums:
            count = 1
            if i - 1 in n_set:
                continue
            else:
                while i + 1 in n_set:
                    count += 1
                    i += 1
            ans = max(ans,count)
            
        return ans